src/component/visualMap/PiecewiseView.ts

```diff
@@ -215,10 +215,13 @@ class PiecewiseVisualMapView extends VisualMapView {
     ) {
         const visualMapModel = this.visualMapModel;
         const option = visualMapModel.option;
+        if (!option.selectedMode) {
```

Comment:
LGTM. A small suggestion on the style: `option.selectedMode` can be assigned to a temporal variable to get a smaller code size.

Reply:
Ok, get it. Thanks for the tip